### PR TITLE
added a generic test

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

File: src/test/java/com/github/roookeee/datus/generic/GenericTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.github.roookeee.datus.api.Datus;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class GenericTest {
```
Comment (wind57) [inline src/test/java/com/github/roookeee/datus/generic/GenericTest.java:11]: not a big deal, but I was looking for a "generic" test while trying some things out

Comment (wind57) [reply src/test/java/com/github/roookeee/datus/generic/GenericTest.java:11]: I am, btw, not aware of a better way to do this. If you take a `Class` as input, but the Class is generic, there is no other way then casting to the types that you actually have. You could change `forTypes` to take `Type`s as arguments, but then compile time safety would be entirely lost. I have faced this problem before and I have no solution to get rid of that `@SuppressWarnings("unchecked")`...

Comment (roookeee) [reply src/test/java/com/github/roookeee/datus/generic/GenericTest.java:11]: Ah tricky, forgot that case too. Will give this some thought and come back to you.

Comment (roookeee) [reply src/test/java/com/github/roookeee/datus/generic/GenericTest.java:11]: My basic idea would be a `<T,U> forTypes()` in which the user specifies the generics in the callsite: `.<Generic<String>, Generic<Int>>forTypes()`. I mean it is pretty ugly but at least it works, type-safe also if I recall correctly. Will try that later :)

Comment (roookeee) [reply src/test/java/com/github/roookeee/datus/generic/GenericTest.java:11]:
```java
package com.github.roookeee.datus.generics;

import com.github.roookeee.datus.api.Datus;
import com.github.roookeee.datus.api.Mapper;
import com.github.roookeee.datus.immutable.ConstructorParameter;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class GenericsTest {

    @Test
    public void shouldSupportGenericTypesInTheImmutableApi() {
        //given
        ImmutableContainer<Integer> input = new ImmutableContainer<>(5);
        Mapper<ImmutableContainer<Integer>, ImmutableContainer<String>> mapper =
                Datus.<ImmutableContainer<Integer>, ImmutableContainer<String>>forTypes()
                // type inference hates us a little here, the generic type of immutable is inferred by the passed lambda
                // which is generic and thus object -> have to specify type although it is a bit ugly
                .<String>immutable(ImmutableContainer::new)
                .from(ImmutableContainer::getElement).map(Object::toString).to(ConstructorParameter::bind)
                .build();

        //when
        ImmutableContainer<String> result = mapper.convert(input);

        //then
        assertThat(result.getElement()).isEqualTo(result.getElement());
    }

    @Test
    public void shouldSupportGenericTypesInTheMutableApi() {
        //given
        MutableContainer<Integer> input = new MutableContainer<>();
        input.setElement(5);
        Mapper<MutableContainer<Integer>, MutableContainer<String>> mapper =
                Datus.<MutableContainer<Integer>, MutableContainer<String>>forTypes()
                        .mutable(MutableContainer::new)
                        .from(MutableContainer::getElement).map(Object::toString).into(MutableContainer::setElement)
                        .build();

        //when
        MutableContainer<String> result = mapper.convert(input);

        //then
        assertThat(result.getElement()).isEqualTo(result.getElement());
    }

    private static class ImmutableContainer<T> {
        private final T element;

        public ImmutableContainer(T element) {
            this.element = element;
        }

        public T getElement() {
            return element;
        }
    }

    private static class MutableContainer<T> {
        private T element;

        public T getElement() {
            return element;
        }

        public void setElement(T element) {
            this.element = element;
        }
    }
}

```

**Comment by roookeee** [comment]
opened #35 for this

**Pull request closed without merging**